comp/core/hostname/hostnameinterface/interface.go

```diff
@@ -0,0 +1,16 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package hostnameinterface describes hostname methods required in logs agent modules
+package hostnameinterface
+
+import (
+	"context"
+)
+
+type HostnameInterface interface {
```

Comment:
I was looking into that interface but the interface needs to be exported as a module in order to be used in `pkg/logs` modules. `component.go` depends on `pkg/util/hostname` which would prevent this from being used

Reply:
The only reason the public interface depends on `pkg/util/hostname` is because of `hostname.Data`. Perhaps that struct can be moved into the public interface definition? 
Something like this: https://github.com/DataDog/datadog-agent/compare/brian/hostname-comp-interface

Reply:
That could work too. I don't really have a preference TBH so long as we are not working around the component design patterns. (it could be worth asking agent-shared-components as they own the hostname component)